src/net_processing.cpp

```diff
@@ -3439,6 +3439,7 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
             }
             if (!fRevertToInv && !vHeaders.empty()) {
                 if (vHeaders.size() == 1 && state.fPreferHeaderAndIDs) {
+                    assert(pBestIndex != nullptr);
```

Comment:
Unrelated to this PR. Above you see that if `vHeaders.push` is called then `pBestIndex = pindex` was called. IMO this should be removed.

Reply:
You don't mean that the assertion should be removed? How do you mean "unrelated to this PR" in this context? :-)

Reply:
This is not an assertion of a LookupBlockIndex result, that's done in L3400.